src/core/services/compiler/compiler.js

```diff
@@ -6,7 +6,21 @@
  */
 angular
   .module('material.core')
-  .service('$mdCompiler', MdCompilerService);
+  .service('$mdCompiler', MdCompilerService)
+  .provider('$$mdPreAssignBindings', PreAssignBindingsProvider);
+
+/**
+ * Provider that is used to report the preAssignBindingsEnabled state to the $mdCompiler service.
+ */
+function PreAssignBindingsProvider($compileProvider) {
+  // To avoid breaking changes in Material, AngularJS Material will only respect the
+  // preAssignBindingsEnabled state in AngularJS 1.6.X and higher.
+  var respectPreAssignState = angular.version.major === 1 && angular.version.minor > 5;
+
+  this.$get = function() {
+    return respectPreAssignState ? $compileProvider.preAssignBindingsEnabled() : true;
```

Comment:
The implementation did not cover the changes we discussed in our meeting. Also we don't want to re-check the `preAssignBindingsEnabled` state for each `$mdCompiler` call.

I changed it now to work as we discussed on our meeting
* 1.5: Bindings will be always pre-assigned as same as with Material before.
* 1.6: Angular Material will start respecting the `preAssignBindingsEnabled` state
* 1.7: Compiler will always sets preAssignBindingsEnabled to false.